components/form/AddNewItemForm.tsx

```diff
@@ -92,6 +95,18 @@ export default function AddNewItemForm({
         ...itemDataWithoutImage,
       };
 
+      try {
+        const addedItem = await insertRow('items', itemData);
+        if (!addedItem || addedItem.length === 0) {
+          throw new Error('Failed to add the new item');
+        }
+        const itemId = addedItem[0].id;
+        reset();
+        router.push(`/add-item/success/${itemId}`);
+      } catch (error) {
+        router.push(`/add-item?message=${error}`);
+      }
```

Comment:
this `try catch` is not needed. We can just push to `/add-item?message='Failed to add the new item'` straight from the `if` statement.

The pattern you used would allow us to pass specific error messages to the user if we were calling an api or something of the sort but in this case we're defining the message ourselves so let's use the early return to keep our logic lean and readable